### ISSUE TITLE
Add check to match FromAsCasing Docker build check

### ISSUE THREAD

**Issue by jnoordsij** [hadolint/hadolint#1026]
- [ ] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Docker has some build checks, which may raise warnings when building images that are incorrect or inconsistent in some way; see also https://docs.docker.com/reference/build-checks/.

The [`FromAsCasing` check](https://docs.docker.com/reference/build-checks/from-as-casing/) tests whether or not the `FROM` and `AS` keywords are consistently using the same casing in any instruction. If this is not consistent, recent Docker versions (tested on 27.0.1) issue a warning ` FromAsCasing: 'as' and 'FROM' keywords' casing do not match`.

However, `hadolint` does not report any error on a Dockerfile violating this rule. I would expect it to do so, given it is a pattern 

### Actual behavior

Building with `docker` reports a warning, `hadolint` does not.

### Steps to reproduce the behavior

1. Create Dockerfile as listed below and run `hadolint`: no warning appears
2. Build Dockerfile with Docker 27.0.1: warning appears

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
hadolint --version
> Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
FROM debian:latest as builder
```

**Additional environment details (OS, stack version, etc.)**

N/A

**Comment by jnoordsij** [comment]
Side note: I haven't checked myself, but there might more build checks that might be suitable candidates to have corresponding checks here?

**Comment by ShoGinn** [comment]
I support this request! The linter should include official Docker build checks or any checks from the official source.

**Comment by mschoettle** [comment]
This would be great to have!

> Side note: I haven't checked myself, but there might more build checks that might be suitable candidates to have corresponding checks here?

Another one that I came across in our projects is [LegacyKeyValueFormat](https://docs.docker.com/reference/build-checks/legacy-key-value-format/).

```Dockerfile
# old
ENV SOME_ENV 1
# new
ENV SOME_ENV=1
```

**Comment by ferrarimarco** [comment]
Hi!

This is the Docker build checks reference that lists all the available checks: https://docs.docker.com/reference/build-checks/
